$DragonFly: src/gnu/lib/gcc34/libstdc++/ctype_base.h.patch,v 1.3 2005/09/17 14:39:44 joerg Exp $

--- ctype_base.h.orig	2005-03-02 01:56:21.000000000 +0100
+++ ctype_base.h	2005-09-16 20:02:33.000000000 +0200
@@ -43,16 +43,16 @@
 
     // NB: Offsets into ctype<char>::_M_table force a particular size
     // on the mask type. Because of this, we don't use an enum.
-    typedef unsigned char      	mask;
-    static const mask upper    	= _U;
-    static const mask lower 	= _L;
-    static const mask alpha 	= _U | _L;
-    static const mask digit 	= _N;
-    static const mask xdigit 	= _N | _X;
-    static const mask space 	= _S;
-    static const mask print 	= _P | _U | _L | _N | _B;
-    static const mask graph 	= _P | _U | _L | _N;
-    static const mask cntrl 	= _C;
-    static const mask punct 	= _P;
-    static const mask alnum 	= _U | _L | _N;
+    typedef uint16_t      	mask;
+    static const mask upper    	= _CTYPEMASK_U;
+    static const mask lower 	= _CTYPEMASK_L;
+    static const mask alpha 	= _CTYPEMASK_A;
+    static const mask digit 	= _CTYPEMASK_D;
+    static const mask xdigit 	= _CTYPEMASK_X;
+    static const mask space 	= _CTYPEMASK_S;
+    static const mask print 	= _CTYPEMASK_R;
+    static const mask graph 	= _CTYPEMASK_G;
+    static const mask cntrl 	= _CTYPEMASK_C;
+    static const mask punct 	= _CTYPEMASK_P;
+    static const mask alnum 	= _CTYPEMASK_A | _CTYPEMASK_D;
   };
